test/Microsoft.AspNet.Routing.Tests/RouteValueDictionaryTests.cs

```diff
@@ -168,6 +168,48 @@ public void CreateFromObject_MixedCaseThrows()
                 expected);
         }
 
+        [Fact]
+        public void CreateFromReadOnlyDictionary_CopiesValues()
+        {
+            // Arrange
+            var dictionary = new Dictionary<string, object>();
+            dictionary.Add("Name", "James");
+            dictionary.Add("Age", 30);
+            var readonlyDictionary = (IReadOnlyDictionary<string, object>)dictionary;
```

Comment:
since the parameter passed to the constructor is always of type `object`, this line makes no difference.  should instead pass a `List<KeyValuePair<string, object>>` or use a state machine method that returns `IEnumerable<KeyValuePair<string, object>>` and uses `yield return`.
same for next test method.